Collector/ProfileClient.php

```diff
@@ -47,23 +47,18 @@ class ProfileClient implements HttpClient, HttpAsyncClient
     private $eventNames = [];
 
     /**
-     * @param HttpClient|HttpAsyncClient $client    The client to profile. Client must implement both HttpClient and
-     *                                              HttpAsyncClient interfaces.
+     * @param HttpClient|HttpAsyncClient $client    The client to profile. Client must implement HttpClient or
+     *                                              HttpAsyncClient interface.
      * @param Collector                  $collector
      * @param Formatter                  $formatter
      * @param Stopwatch                  $stopwatch
      */
     public function __construct($client, Collector $collector, Formatter $formatter, Stopwatch $stopwatch)
     {
         if (!($client instanceof HttpClient && $client instanceof HttpAsyncClient)) {
-            throw new \RuntimeException(sprintf(
```

Comment:
Shouldn't the exception still be thrown if neither of the interfaces is implemented? I know that the `FlexibleHttpClient` throws anyway, but I am not sure if relying on that behaviour would be a good idea and also could the stack trace be confusing for users.

Reply:
Yes, an exception are thrown, But yes. It could be a good idea to throw en exception here as well.